src/ripple/net/SSLHTTPDownloader.h

```diff
@@ -65,8 +65,10 @@ class SSLHTTPDownloader
 private:
     boost::asio::ssl::context ctx_;
     boost::asio::io_service::strand strand_;
-    boost::asio::ssl::stream<boost::asio::ip::tcp::socket> stream_;
+    boost::optional<
```

Comment:
This is made optional but it's never checked anywhere for being valid, all those places would throw `bad_optional_access`. Then on clean-up it's set to `none`, is it to catch use-after-free errors?